temporalio/worker/replayer.py

```diff
@@ -0,0 +1,272 @@
+"""Replayer."""
+
+from __future__ import annotations
+
+import concurrent.futures
+import copy
+import json
+import logging
+import re
+from typing import Any, Dict, Iterable, Optional, Type, Union
+
+import google.protobuf.json_format
+from typing_extensions import TypedDict
+
+import temporalio.api.history.v1
+import temporalio.bridge.worker
+import temporalio.converter
+
+from .interceptor import Interceptor
+from .worker import load_default_build_id
+from .workflow import _WorkflowWorker
+from .workflow_instance import UnsandboxedWorkflowRunner, WorkflowRunner
+
+logger = logging.getLogger(__name__)
+
+
+class Replayer:
+    """Replayer to replay workflows from history."""
+
+    def __init__(
+        self,
+        *,
+        workflows: Iterable[Type],
+        workflow_task_executor: Optional[concurrent.futures.ThreadPoolExecutor] = None,
+        workflow_runner: WorkflowRunner = UnsandboxedWorkflowRunner(),
+        namespace: str = "ReplayNamespace",
+        data_converter: temporalio.converter.DataConverter = temporalio.converter.default(),
+        interceptors: Iterable[Interceptor] = [],
+        build_id: Optional[str] = None,
+        identity: Optional[str] = None,
+        debug_mode: bool = False,
+    ) -> None:
+        """Create a replayer to replay workflows from history.
+
+        See :py:meth:`temporalio.worker.Worker.__init__` for a description of
+        arguments.
+        """
+        if not workflows:
+            raise ValueError("At least one workflow must be specified")
+        self._config = ReplayerConfig(
+            workflows=workflows,
```

Comment:
Hmm.. wouldn't you want the workflows to be turned into a list here so if you get an iterator you can re-read it?

Reply:
Hrmm, yeah I take iterables in several places and just pass them around and read them once. I do need to wrap as a list in all of those places. I have opened https://github.com/temporalio/sdk-python/issues/110, but I will fix this one in the meantime.

Reply:
Fixed